app/controllers/bookmarks_controller.rb

```diff
@@ -26,12 +43,22 @@ def csv
 
   private
 
-    def fetch_bookmarked_documents
+    def bookmark_ids
       bookmarks = token_or_current_or_guest_user.bookmarks
       bookmark_ids = bookmarks.collect { |b| b.document_id.to_s }
+      bookmark_ids + alma_ids(bookmark_ids)
+    end
+
+    def fetch_bookmarked_documents
       _, @documents = search_service.fetch(bookmark_ids, rows: bookmark_ids.length, fl: '*')
     end
 
+    def alma_ids(bookmark_ids)
+      bookmark_ids.map do |id|
+        "99#{id}3506421"
+      end
+    end
+
     # byte-order-mark declaring our output as UTF-8 (required for non-ASCII to be handled by Excel)
```

Comment:
This Pr looks ok for the voyager ids. I have a question for @mzelesky about the SCSB ID extraction. If the princeton recap ids are different then we would need to consider this transformation as well for the bookmarks. But I'm not sure.

Reply:
@mzelesky said that the Ids of the Princeton records we send to scsb will not change.